--- a/Makefile	2012-01-27 18:00:00.000000000 +0100
+++ b/Makefile	2012-01-27 18:00:43.750701648 +0100
@@ -158,14 +158,6 @@
 configcheck: sanitycheck
 	@echo -n "Checking kernel configuration... "
 	
-	@# check version of kernel
-	@echo $(KERNELRELEASE) | grep -q -i '^[2-9]\.[4-9]\.' || { \
-	    echo "FAILED"; \
-	    echo "Only kernel versions 2.4.x and above are supported."; \
-	    echo "You have $(KERNELRELEASE)."; \
-	    exit 1; \
-	}
-	
 	@# check kernel configuration
 	@if [ -z "$(CONFIG_SYSCTL)" ]; then \
 	    echo "FAILED"; \
